log/logrus/logrus_logger_test.go

```diff
@@ -53,3 +55,66 @@ func TestLogrusLogger(t *testing.T) {
 type mymap map[int]int
 
 func (m mymap) String() string { return "special_behavior" }
+
+func TestWithLevel(t *testing.T) {
+	tests := []struct {
+		name          string
+		level         logrus.Level
+		expectedLevel logrus.Level
+	}{
+		{
+			name:          "Test Debug level",
+			level:         logrus.DebugLevel,
+			expectedLevel: logrus.DebugLevel,
+		},
+		{
+			name:          "Test Error level",
+			level:         logrus.ErrorLevel,
+			expectedLevel: logrus.ErrorLevel,
+		},
+		{
+			name:          "Test Warn level",
+			level:         logrus.WarnLevel,
+			expectedLevel: logrus.WarnLevel,
+		},
+		{
+			name:          "Test Info level",
+			level:         logrus.InfoLevel,
+			expectedLevel: logrus.InfoLevel,
+		},
+		{
+			name:          "Test Trace level",
+			level:         logrus.TraceLevel,
+			expectedLevel: logrus.TraceLevel,
+		},
+		{
+			name:          "Test not existing level",
+			level:         999,
+			expectedLevel: logrus.InfoLevel,
+		},
+	}
+	for _, tt := range tests {
+		buf := &bytes.Buffer{}
+		logrusLogger := logrus.New()
+		logrusLogger.Out = buf
+		logrusLogger.Level = tt.level
+		logrusLogger.Formatter = &logrus.JSONFormatter{}
+		logger := log.NewLogger(logrusLogger, log.WithLevel(tt.level))
+
+		t.Run(tt.name, func(t *testing.T) {
+			if err := logger.Log(time.Now().String(), "amazing", "tests"); err != nil {
```

Comment:
This `Log` call seems to be missing either a key or a value. I also wonder why pass these keyvals here when they aren't checked later. I don't think they are needed to test the `WithLevel` functionality, so maybe it's more clear to just remove them.